src/Symfony/Component/Translation/Extractor/PhpAstExtractor.php

```diff
@@ -64,7 +69,15 @@ public function setPrefix(string $prefix): void
 
     protected function canBeExtracted(string $file): bool
     {
-        return 'php' === pathinfo($file, \PATHINFO_EXTENSION) && $this->isFile($file);
+        $content = file_get_contents($file);
+
+        return 'php' === pathinfo($file, \PATHINFO_EXTENSION)
+            && $this->isFile($file)
+            && ((bool) preg_match(self::T_FUNCTION_REGEX, $content)
```

Comment:
```suggestion
            && ((bool) preg_match(self::T_FUNCTION_REGEX, $content = file_get_contents($file))
```
To avoid getting the content if the file is not a PHP file?